Assets/LoggingManager/LogStore.cs

```diff
@@ -0,0 +1,276 @@
+            if (createStringOverTime)
+            {
+                if (currentLineLogged.Length != 0)
+                {
+                    currentLineLogged.Append(fieldSeparator);
+                }
+                currentLineLogged.Append(pair.Value);
+            }
+        }
+
+        if (createStringOverTime)
+        {
+            currentLineLogged.Append(lineSeperator);
+            logString.Append(currentLineLogged);
+            currentLineLogged.Clear();
+        }
+        currentLogs.Clear();
+        nbLines++;
+    }
+
+    //Clears all the logs
+    public void Clear()
+    {
+        logs.Clear();
+        currentLogs.Clear();
+        logString.Clear();
+        nbLines = 0;
+    }
+
+    //Exports the logs into a string or into a Dictionarry<string,string>
+    public T ExportAll<T>()
+    {
+        if (currentLogs.Count != 0)
+        {
+            TerminateRow();
+        }
+        var type = typeof(T);
+        if (type == typeof(Dictionary<string, string>))
+        {
+            return (T)Convert.ChangeType(logs, type);
+        }
+        if (type == typeof(string))
+        {
+            return (T)Convert.ChangeType(ExportToString(), type);
+        }
+        throw new RuntimeException("Export type must be a Dictionnary<string,string> or string");
+    }
+
+    //Exports the logs to a string
+    private string ExportToString()
+    {
+        if (!createStringOverTime)
+        {
+            logString.Clear();
+            for (int i = 0; i < nbLines; i++)
+            {
+                string line = "";
+                foreach (string key in logs.Keys)
+                {
+                    if (line != "")
+                    {
+                        line += fieldSeparator;
+                    }
+                    line += logs[key][i];
+                }
+
+                logString.Append(line + lineSeperator);
+            }
+        }
+        return logString.ToString();
+    }
+
+    //Returns the headers as a string 
+    public string GenerateHeaders()
+    {
+        string headers = "";
+        foreach (string key in logs.Keys)
+        {
+            if (headers != "")
+            {
+                headers += fieldSeparator;
+            }
+            headers += key;
+        }
+        return headers;
+    }
+
+
+    // Converts the values of the parameters (in a "object format") to a string, formatting them to the
+    // correct format in the process.
+    private static string ConvertToString(object arg)
+    {
+        if (arg is float)
+        {
+            return ((float)arg).ToString("0.0000").Replace(",", ".");
+        }
+        if (arg is int)
+        {
+            return arg.ToString();
+        }
+        if (arg is bool)
+        {
+            return ((bool)arg) ? "TRUE" : "FALSE";
+        }
+        if (arg is Vector3)
+        {
+            return ((Vector3)arg).ToString("0.0000").Replace(",", ".");
+        }
+        return arg.ToString();
+    }
```

Comment:
Do we have a sanitization process e.g. to remove special characters like `;` or `\n` (field separator, etc) inside `LogStore` ? Or is this done by the LoggingManager? Maybe its better to do this in LogStore, perhaps as part of this function. What do you think?

Reply:
I think it's a good idea to do this in LogStore.